src/Microsoft.DncEng.AIChatBot/backend/Services/Skills/UpdateContextVariableSkill.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+namespace MinimalApi.Services.Skills;
+
+public sealed class UpdateContextVariableSkill
+{
+    [SKFunction("Update knowledge")]
+    [SKFunctionName("UpdateKnowledgeVariable")]
+    [SKFunctionInput(Description = "The value to add or append")]
+    [SKFunctionContextParameter(Name = "knowledge", Description = "variable to update")]
+    public void AddOrAppend(string variableValue, SKContext context)
+    {
+        if (context.Variables.ContainsKey("knowledge"))
+        {
+            context.Variables["knowledge"] = $"{context.Variables["knowledge"]}\r{variableValue}";
+        }
+        else
+        {
+            context.Variables["knowledge"] = variableValue;
+        }
```

Comment:
nit: I think it's better to do this:
```suggestion
        if (context.Variables.TryGetValue("knowledge", out var value))
        {
            variableValue = $"{value}\r{variableValue}";
        }

        context.Variables["knowledge"] = variableValue;
```